Owncloud iOs Client/Tabs/FileTab/FilesViewController.m

```diff
@@ -1831,23 +1832,25 @@ -(void)reloadTableFromDataBase {
 
 
 -(void)reloadTableFileListAfterCapabilitiesUpdated {
-    
-    [self updateCurrentFileToShowIfNeeded];
-    
-    [self reloadTableFromDataBase];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self updateCurrentFileToShowIfNeeded];
+        [self reloadTableFromDataBase];
+    });
 }
 
 - (void) updateCurrentFileToShowIfNeeded {
-    AppDelegate *app = (AppDelegate*)[[UIApplication sharedApplication] delegate];
-    
-    //Workaround to find the _fileIdToShowFiles because some times there are problems with the changes active user, and this method is launched before the viewwillappear
-    if (app.activeUser) {
-        FileDto *rootFileDtoOfActiveUser = [ManageFilesDB getRootFileDtoByUser:app.activeUser];
-        if (self.fileIdToShowFiles.userId != rootFileDtoOfActiveUser.userId) {
-            _fileIdToShowFiles = rootFileDtoOfActiveUser;
-            DLog(@"Changing between accounts, update _fileIdToShowFiles with root file of active user %ld, %@", (long)app.activeUser.userId, app.activeUser.username);
+    dispatch_async(dispatch_get_main_queue(), ^{
+        AppDelegate *app = (AppDelegate*)[[UIApplication sharedApplication] delegate];
+        
+        //Workaround to find the _fileIdToShowFiles because some times there are problems with the changes active user, and this method is launched before the viewwillappear
+        if (app.activeUser) {
+            FileDto *rootFileDtoOfActiveUser = [ManageFilesDB getRootFileDtoByUser:app.activeUser];
```

Comment:
Here you use again the appDelegate. I think it's better not use this and use the database method `getActiveUser`